docs/API.md

```diff
@@ -633,16 +638,50 @@ An object with the following properties:
 
 #### `"result"`
 
-A *test id*. 
+A *test id*.
 
-If the test has been run with the same domain name within an interval of 10 mins (hard coded), 
+If the test has been run with the same domain name within an interval of 10 mins (hard coded),
 then the new request does not trigger a new test, but returns with the results of the last test
 
 
 #### `"error"`
 
-* If the given `profile` is not among the [available profiles], a user
-  error is returned.
+* If any parameter fails to be validated an error code of -32602 is returned. The `data` properties contains an array of all error.
+
+  Example of error response:
+
+```json
+{
+  "error": {
+    "code": "-32602",
+    "data": [
+      {
+        "message": "Expected integer - got string.",
+        "path": "/ds_info/0/algorithm"
+      },
+      {
+        "message": "Missing property.",
+        "path": "/ds_info/0/digest"
+      },
+      {
+        "path": "/profile",
+        "message": "Unknown profile"
+      },
+      {
+        "path": "/domain",
+        "message": "The domain name character(s) are not supported"
+      },
+      {
+        "path": "/nameservers/0/ip",
+        "message": "Invalid IP address"
+      }
+    ],
+    "message": "Invalid method parameter(s)."
+  },
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
```

Comment:
Could the `data` property be formally defined?

Reply:
For validation errors we should have a generic formal description somewhere at the top.

Regarding the old text, I think that came about because the API documentation doesn't contain all the information required to fully validate the profile names. The *profile name* data type specifies the general pattern in terms of a regular expression. But the check against the configuration wasn't performed by the validation layer at all, but rather by the method itself. (I haven't had time to look at what the changes in this PR yet, only the discussions.)